### PR TITLE
Fix RDS test suite and minor bugs revealed

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY

* Update testing policy to be correct for RDS test suite
* Fix Iops bug when Iops is not present
* Create read replica in same region to avoid more permissions being
  required
* Ensure modifying DB doesn't try to downgrade engine version
* Add tags to main test suite to limit number of tests run for problem
  solving

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
rds_instance

### CODE REVIEW

**Comment by willthames** [comment]
Based on https://github.com/ansible/ansible/pull/57716 - merge that one first

File: lib/ansible/modules/cloud/amazon/rds_instance.py [hunk 1]
```diff
@@ -847,7 +847,7 @@ def get_options_with_changing_values(client, module, parameters):
     if cloudwatch_logs_enabled:
         parameters['CloudwatchLogsExportConfiguration'] = cloudwatch_logs_enabled
     if not module.params['storage_type']:
-        parameters.pop('Iops')
+        parameters.pop('Iops', None)
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/rds_instance.py]: I'm wondering if you could separate this so it can be backported.

Comment (willthames) [reply lib/ansible/modules/cloud/amazon/rds_instance.py]: yeah, of course

Comment (willthames) [reply lib/ansible/modules/cloud/amazon/rds_instance.py]: I've taken it out of this PR, once my currently running rds test suite is passing (I'm pretty confident it's all good now) I'll cherry pick onto a new branch

Comment (s-hertel) [reply lib/ansible/modules/cloud/amazon/rds_instance.py]: Thanks so much

**Review by s-hertel — APPROVED** [review]
Looks good and the tests pass for me. Thanks for the painstaking hours!

**Pull request merged**